packages/mapviewer/src/modules/menu/components/activeLayers/MenuActiveLayersList.vue

```diff
@@ -46,7 +47,27 @@ onMounted(() => {
         onEnd: function (event) {
             aLayerIsDragged.value = false
             const { newIndex, oldIndex } = event
-            onMoveLayer(reverseIndex(oldIndex), reverseIndex(newIndex))
+            if (
+                newIndex >= 0 &&
+                newIndex < activeLayers.value.length &&
+                oldIndex >= 0 &&
+                oldIndex < activeLayers.value.length
+            ) {
+                nextTick(() => {
+                    // Get all children with draggable="false" and remove them
+                    const nonDraggableChildren =
+                        activeLayersList.value.querySelectorAll('[draggable="false"]')
+                    if (nonDraggableChildren.length > 0) {
+                        nonDraggableChildren.forEach((child) => child.remove())
+                        log.debug('Non-draggable children removed:', nonDraggableChildren)
+                    } else {
+                        log.debug('No non-draggable children found')
+                    }
+                    onMoveLayer(reverseIndex(oldIndex), reverseIndex(newIndex))
+                })
```

Comment:
works ok. nice find!
can you add a little comment to explain why we are doing this search and destroy on draggable=false?
Just make it clear that is is to fix this bug so we don't remove it in the future because we don't understand why 😉